examples/ui/ui_material.rs

```diff
@@ -57,12 +56,39 @@ fn setup(mut commands: Commands, mut ui_materials: ResMut<Assets<CustomUiMateria
 
 #[derive(AsBindGroup, Asset, TypePath, Debug, Clone)]
 struct CustomUiMaterial {
+    /// Color multiplied with the image
     #[uniform(0)]
     color: Vec4,
+    /// Represents how much of the image is visible
+    /// Goes from 0 to 1
+    #[uniform(1)]
+    slider: f32,
+    /// Image used to represent the slider
+    #[texture(2)]
+    #[sampler(3)]
+    color_texture: Handle<Image>,
 }
 
 impl UiMaterial for CustomUiMaterial {
     fn fragment_shader() -> ShaderRef {
         SHADER_ASSET_PATH.into()
     }
 }
+
+// Fills the slider slowly over 2 seconds and resets it
+// Also updates the color of the image to a rainbow color
+fn animate(
+    mut materials: ResMut<Assets<CustomUiMaterial>>,
+    q: Query<&Handle<CustomUiMaterial>>,
+    time: Res<Time>,
+) {
+    let duration = 2.0;
+    for handle in &q {
+        if let Some(material) = materials.get_mut(handle) {
+            // rainbow color effect
+            let new_color = Color::hsl((time.elapsed_seconds() * 60.0) % 360.0, 1., 0.5);
+            material.color = LinearRgba::from(new_color).to_f32_array().into();
+            material.slider = (time.elapsed_seconds() % duration) / duration;
```

Comment:
I think it would be prettier with

```suggestion
            material.slider = ((time.elapsed_seconds() % (duration * 2.0)) - duration).abs() / duration;
```